File: Selenium/Project/PrActivity7.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager

# Set up Firefox driver
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Navigate to the OrangeHRM login page
    driver.get("http://alchemy.hguy.co/orangehrm")

    # Find and enter the username
    driver.find_element(By.ID, "txtUsername").send_keys("orange")
    # Find and enter the password
    driver.find_element(By.ID, "txtPassword").send_keys("orangepassword123")
    # Click on the login button
    driver.find_element(By.ID, "btnLogin").click()

    # Click on My Info menu item
    driver.find_element(By.ID, "menu_pim_viewMyDetails").click()

    # Click on Qualifications option on the left side menu
    driver.find_element(By.LINK_TEXT, "Qualifications").click()

    # Click on Add button
    driver.find_element(By.ID, "addWorkExperience").click()

    # Fill in the Work Experience form
    driver.find_element(By.ID, "experience_employer").send_keys("IBM India Pvt Ltd")
    driver.find_element(By.ID, "experience_jobtitle").send_keys("Software Engineer")
    driver.find_element(By.ID, "experience_from_date").send_keys("2020-01-01")
    driver.find_element(By.ID, "experience_to_date").send_keys("2024-06-01")
    driver.find_element(By.ID, "experience_comments").send_keys("Worked as a software engineer.")

    # Click Save
    driver.find_element(By.ID, "btnWorkExpSave").click()

# Close the browser
driver.quit()

File: Selenium/Project/PrActivity7.java
package Project;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.testng.annotations.*;

public class PrActivity7 {
    WebDriver driver;

    @BeforeClass
    public void setup() {
        // Set up the Firefox driver
        WebDriverManager.firefoxdriver().setup();
        driver = new FirefoxDriver();
    }

    @Test
    public void addQualifications() {
        // Open OrangeHRM page
        driver.get("http://alchemy.hguy.co/orangehrm");

        // Login with credentials
        WebElement username = driver.findElement(By.id("txtUsername"));
        username.sendKeys("orange");

        WebElement password = driver.findElement(By.id("txtPassword"));
        password.sendKeys("orangepassword123");

        WebElement loginBtn = driver.findElement(By.id("btnLogin"));
        loginBtn.click();

        // Click on My Info menu
        WebElement myInfoMenu = driver.findElement(By.id("menu_pim_viewMyDetails"));
        myInfoMenu.click();

        // Click on Qualifications option
        WebElement qualificationsOption = driver.findElement(By.xpath("//ul[@id='sidenav']//a[contains(text(),'Qualifications')]"));
        qualificationsOption.click();

        // Click on Add Work Experience
        WebElement addWorkExperienceBtn = driver.findElement(By.id("addWorkExperience"));
        addWorkExperienceBtn.click();

        // Fill in Work Experience details
        WebElement companyName = driver.findElement(By.id("experience_employer"));
        companyName.sendKeys("ABC Company");

        WebElement jobTitle = driver.findElement(By.id("experience_jobtitle"));
        jobTitle.sendKeys("Software Engineer");

        WebElement fromDate = driver.findElement(By.id("experience_from_date"));
        fromDate.sendKeys("2024-01-01");

        WebElement toDate = driver.findElement(By.id("experience_to_date"));
        toDate.sendKeys("2024-12-31");

        // Save the work experience
        WebElement saveBtn = driver.findElement(By.id("btnWorkExpSave"));
        saveBtn.click();
    }

    @AfterClass
    public void tearDown() {
        // Close the browser
        driver.quit();
    }
}

